lib/prawn/embedded_files/embedded_file.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'digest/md5'
+
+module Prawn
+  module EmbeddedFiles
+    class EmbeddedFile
+      attr_reader :chksum
+
+      def initialize(data, options = {})
+        @creation_date = options[:creation_date]
+        unless @creation_date.kind_of?(Time)
+          @creation_date = Time.now.utc
+        end
+
+        @mod_date = options[:modification_date]
+        unless @mod_date.kind_of?(Time)
+          @mod_date = Time.now.utc
+        end
+
+        @chksum = Digest::MD5.digest(data)
+        @data = data
+      end
+
+      def build_pdf_object(document)
+        obj = document.ref!(
+          Type: :EmbeddedFile,
+          Params: {
+            CreationDate: @creation_date,
+            ModDate: @mod_date,
+            CheckSum: PDF::Core::LiteralString.new(@chksum),
+            Size: @data.length
+          }
+        )
+
+        obj << @data
+        obj.stream.compress!
```

Comment:
Document has a global option that controls compression so we should honour it here. This should be behind `if document.compression_enabled?` guard.